File: files/2025-05-18/CSEC 18052025.R
library(tidyverse)
library(readxl)

path = "files/2025-05-18/Challenge24.xlsx"
start = 2
end = 7
skip = 4
repetition = 3
test = read_excel(path, range = "G3:G18")

result = data.frame(List = rep(setdiff(start:end, skip), each = repetition))

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/2025-05-18/CSEC 18052025.py
import pandas as pd

path = "files/2025-05-18/Challenge24.xlsx"
start, end, skip, repetition = 2, 7, 4, 3
test = pd.read_excel(path, usecols="G", skiprows=2, nrows=16)

lst = [i for i in range(start, end+1) if i != skip] * repetition
result = pd.DataFrame({'List': sorted(lst)})

print(result.equals(test))